Update Lambda OOM metric doc (#31092)
DataDog/documentation: content/en/serverless/aws_lambda/metrics.md

`aws.lambda.enhanced.out_of_memory`
: Measures the number of times a function runs out of memory.

> : Due to the many variations of out-of-memory errors, some cases may not be handled well despite our best efforts. If you encounter such a case, please create an issue in the [Datadog Lambda Extension GitHub repo][9].

Reviewer: ```suggestion
: Because of the many variations of out-of-memory errors, some cases may not be handled well despite our best efforts. If you encounter such a case, create an issue in the [Datadog Lambda Extension GitHub repo][9].
```
